proxy.go

```diff
@@ -40,15 +40,18 @@ const (
 //
 // Before creating a Proxy, users must set any socket options,
 // and Listen or Dial both frontend and backend sockets.
-func NewProxy(ctx context.Context, front, back, capture Socket) *Proxy {
+func NewProxy(ctx context.Context, front, back, capture Socket) (*Proxy, error) {
 	grp, ctx := errgroup.WithContext(ctx)
 	proxy := Proxy{
 		ctx:  ctx,
 		grp:  grp,
 		cmds: make(chan proxyCmd),
 	}
-	proxy.init(front, back, capture)
-	return &proxy
+	if err := proxy.init(front, back, capture); err != nil {
+		return nil, err
+	} else {
+		return nil, err
+	}
```

Comment:
this should probably read instead:
```suggestion
	if err := proxy.init(front, back, capture); err != nil {
		return nil, err
	}
	return proxy, nil
```

Reply:
That's what happens when I work too quickly. Committed.